### DISCUSSION TITLE
example

### DISCUSSION THREAD

**Discussion by dbones** [dotnet/dotnet-operator-sdk#387] [Q&A]
is there an end-to-end example, let how we can write one of these to create a database in postgres?

something which shows 

- setup up of the .net project
- create an entity
- create a controller (creats a database in on a server, with a name, sync the name if it changes)
- build and test
- deploy the operator
- deploy a manifest which uses the CRD

the above is something i would find really helpful (just to get started)

**Comment by buehler** [comment]
Hey @dbones 

Not yet. However, there is a `dotnet new` template that creates a full operator with all operator-related elements (entity, controller, finalizer, webhook).

In short, I can provide the following:
- setting up the project and creating the operator with entities and stuff: use the template
- build: normal dotnet build (I prefer "nuke" to build my stuff as can be viewed in https://github.com/buehler/dotnet-operator-sdk/tree/master/build)
- test: have a look at https://github.com/buehler/dotnet-operator-sdk/tree/master/tests/KubeOps.TestOperator.Test
- deploy: the operator creates the needed installation configs (yaml) in "kustomize" yaml files in the config directory. so to install the operator, one can just reference the kustomization yaml file as seen in https://github.com/WirePact/k8s-operator/blob/main/WirePact.Operator/config/install/kustomization.yaml and https://github.com/WirePact/k8s-demo-application/blob/main/prerequisities/kustomization.yaml. The "wirepact demo app" installs the operator via kustomize.
- Deploy a manifest: also in the https://github.com/WirePact/k8s-demo-application/blob/main/prerequisities/kustomization.yaml this uses the CRD "translator" in the installation itself

I hope that these information may help you to get started.
